addons/l10n_id_pos/models/pos_payment_method.py

```diff
@@ -0,0 +1,23 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+import pytz
+from odoo import fields, models
+
+
+class PosPaymentMethod(models.Model):
+    _inherit = "pos.payment.method"
+
+    def l10n_id_verify_qris_status(self, trx_uuid):
+        """ Verify qris payment status from the provided transaction UUID
+
+        For all qris_invoice_details linked to the transaction, check the payment status
+        """
+        if self.payment_method_type != 'qr_code' or self.qr_code_method != 'id_qr':
+            return True
+        trx = self.env['l10n_id.qris.transaction'].search([('model', '=', 'pos.order'), ('model_id', '=', trx_uuid)])
+        if not trx:
+            return False
```

Comment:
Technically it's impossible since to arrive at the option of confirming payment, QR has to be generated first and therefore a transaction record must be created.
I'm putting guard here just in case :sweat_smile:

Reply:
Wouldn't it make more sense to raise an error then?
I wonder how it would look like for the POS user: they wouldn't be notified that something broke right?

Reply:
indeed, will raise exception here